src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -74,56 +76,74 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
                 return false;
             }
 
-            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Formatting_document))
-            {
-                var cancellationToken = context.OperationContext.UserCancellationToken;
-
-                var docOptions = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
-
-                using (var transaction = new CaretPreservingEditTransaction(
-                    EditorFeaturesResources.Formatting, args.TextView, _undoHistoryRegistry, _editorOperationsFactoryService))
+            context.OperationContext.TakeOwnership();
+            _waitIndicator.Wait(
```

Comment:
@dpoeschl is this new editor command handler thing?

Reply:
can It just get scope from AddScope and change its progress and description? rather than completely take over ownership and do our own stuff with our own wait indicator?

Reply:
where this _waitIndicator came from?

Reply:
> is this new editor command handler thing?

yes, it is.

> can It just get scope from AddScope and change its progress and description? rather than completely take over ownership and do our own stuff with our own wait indicator?

Not currently.  See https://github.com/dotnet/roslyn/issues/28354

The issue is that the current impl of CommentExecutionContext is simply broken currently.  So you cannot properly use it to display things like progress (or ever give an updated message).  Unless it is fixed, it's not really ready for prime time unfortunately.